### ISSUE TITLE
aria-conditional-attr correct for checkbox, but silent on radios

### ISSUE THREAD

**Issue by scottaohara** [dequelabs/axe-core#4416]
The aria-conditional-attr  correctly identifies when the aria-checked attribute is used on a native checkbox - and while that's the primary example that had been discussed per the creation of this rule, the same failure should be called out if someone does `<input type=radio aria-checked=...>` since it results in the same situation of the ARIA attribute not necessarily matching the state of the html input.

The bit we added to ARIA in HTML about this was worded to be for both input types:
>Authors MUST NOT use the [aria-checked](https://www.w3.org/TR/wai-aria-1.2/#aria-checked) attribute on any element where the [checkedness](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-checked), or the indeterminate checked value of the element can be in opposition to the current value of the aria-checked attribute.
